### DISCUSSION TITLE
Behaviour of `goto_file` triggered by key combination `gf`, please help decide!

### DISCUSSION THREAD

**Discussion by typetetris** [helix-editor/helix#2636] [General] [answered]
Hi!

I am trying to improve helix `gf` behaviour.

See #2607 

So far I have implemented:

- If the selected path is relative, it will be resolved relative to the current documents directory. Like in Vim and extremely helpful for [Nix](https://nixos.org) for example.
- If the selection is a single character, use textobjects to get the word under the cursor, so you get the Word the cursor is "in", even if the selection is the first character of the word. (Former behaviour was taking the previous word in this case.)
- If the selection contains a directory path, open the file picker for that directory.

There are two cases where @pickfire (Thanks for the review!) and I couldn't agree. Please share your opinion on following matters:

1. If the selection contains only whitespace or is empty (is that possible), should it (a) do nothing or (b) open the file picker for the directory of the current document?
2. If the selection contains a path not existing in the file system, should it (a) print an error message (like in vim) or (b) open a buffer for the path or even create it?

I would probably open a poll for each question after some discussion. Maybe someone else has entirely different ideas?

Yours sincerely,

Eric

**Comment by pickfire** [comment]
In my personal opinion, 1a, 2b (without creating the file).

**Comment by typetetris** [comment]
Seems like no one cares. How to advance?

**Reply by pickfire** [reply]
I think the other maintainers are probably too busy and missed this, given that there are so many open pull requests and issues to be reviewed. Maybe you can try discuss this in matrix chat, not sure if archseer have any opinion on this.

I did give my opinion on this and I think the answer is general enough, what's your opinion and how is it different from what I suggested? If both of us can come to a common viewpoint then I am sure it should be acceptable by others, then we don't have to ping them.

**Comment by CBenoit** [comment, marked as answer]
Same opinion as pickfire: 1a, 2b but without creating the file (requires to actually `:write` it before creating any file)

**Reply by pickfire** [reply]
@typetetris I guess we can advance with this unless you think it isn't a good idea then you can say why.

**Comment by typetetris** [comment]
I am on vacation. Will come back to this afterwards.

Yours sincerely, Eric


Ivan Tham ***@***.***> schrieb am Mi., 8. Juni 2022, 02:12:

**Comment by rsHalford** [comment]
Hopefully, this is something that is still being considered. I would vote for 1a and 2b. But, I would like to add another level to 2b with either i) adds file extension from parent if not already there ii) just the original highlighted text as file name.

I'd prefer 2bi, as a way to create markdown `[[wikilinks]]`, which when highlighting the word _wikilinks_ and entering `gf` will create a relative file of `wikilinks.md`.

**Comment by sheyll** [comment]
1b and 2b

Also, thank you @typetetris, I really appreciate that you try to fix the `gf` behaviour, since I prefer helix over vim in many regards, but for what I currently do, i.e. editing `.nix` files, a properly (i.e. like vim) functioning `gf` command is really important, since file inclusion is an important aspect modularity when using Nix.
